### DISCUSSION TITLE
How to add barcode functionality in serenity for products?

### DISCUSSION THREAD

**Discussion by imhemayatsangin** [serenity-is/Serenity#5957] [Help] [answered]
Suppose I'm going to develop a Point of Sales app using serenity which will have the following functionality.
1. Creating barcodes while adding new product 
2. detecting multiple products by barcode reader one by one while creating invoice or selling products
3. Printing products barcodes for those products which we have created in the system.

Please suggest me how to do and implement this idea if anyone has done this then please share the source code via my email, I will be so much thankful.
Regards,

**Comment by VictorTomaili** [comment, marked as answer]
1. Creating barcodes may need some additional libraries. Serenity doesn't have any. You can implement it like any other web application.
2. Barcode readers are behaving like a keyboard. You don't need to do anything in web applications. You just need to focus on the barcode field. 
3. Printing barcode is similar to print any other paper. Just use build-in browser print behavior. If you don't want, you also can develop an application what talks with the printer and web application server.

**Reply by imhemayatsangin** [reply]
Thanks Victor Bro.
